x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/indexlifecycle/action/TransportExplainLifecycleAction.java

```diff
@@ -64,20 +65,21 @@ protected void doMasterOperation(ExplainLifecycleRequest request, String[] concr
         for (String index : concreteIndices) {
             IndexMetaData idxMetadata = state.metaData().index(index);
             Settings idxSettings = idxMetadata.getSettings();
+            LifecycleExecutionState lifecycleState = LifecycleExecutionState.fromIndexMetadata(idxMetadata);
             String policyName = LifecycleSettings.LIFECYCLE_NAME_SETTING.get(idxSettings);
             final IndexLifecycleExplainResponse indexResponse;
             if (Strings.hasLength(policyName)) {
                 indexResponse = IndexLifecycleExplainResponse.newManagedIndexResponse(index, policyName,
-                        LifecycleSettings.LIFECYCLE_SKIP_SETTING.get(idxSettings),
-                        LifecycleSettings.LIFECYCLE_INDEX_CREATION_DATE_SETTING.get(idxSettings),
-                        LifecycleSettings.LIFECYCLE_PHASE_SETTING.get(idxSettings),
-                        LifecycleSettings.LIFECYCLE_ACTION_SETTING.get(idxSettings),
-                        LifecycleSettings.LIFECYCLE_STEP_SETTING.get(idxSettings),
-                        LifecycleSettings.LIFECYCLE_FAILED_STEP_SETTING.get(idxSettings),
-                        LifecycleSettings.LIFECYCLE_PHASE_TIME_SETTING.get(idxSettings),
-                        LifecycleSettings.LIFECYCLE_ACTION_TIME_SETTING.get(idxSettings),
-                        LifecycleSettings.LIFECYCLE_STEP_TIME_SETTING.get(idxSettings),
-                        new BytesArray(LifecycleSettings.LIFECYCLE_STEP_INFO_SETTING.get(idxSettings)));
+                    LifecycleSettings.LIFECYCLE_SKIP_SETTING.get(idxSettings),
+                    lifecycleState.getIndexCreationDate() == null ? -1 : lifecycleState.getIndexCreationDate(),
+                    lifecycleState.getPhase() == null ? "" : lifecycleState.getPhase(),
+                    lifecycleState.getAction() == null ? "" : lifecycleState.getAction(),
+                    lifecycleState.getStep() == null ? "" : lifecycleState.getStep(),
+                    lifecycleState.getFailedStep() == null ? "" : lifecycleState.getFailedStep(),
+                    lifecycleState.getPhaseTime() == null ? -1 : lifecycleState.getPhaseTime(),
+                    lifecycleState.getActionTime() == null ? -1 : lifecycleState.getActionTime(),
+                    lifecycleState.getStepTime() == null ? -1 : lifecycleState.getStepTime(),
+                    new BytesArray(lifecycleState.getStepInfo() == null ? "" : lifecycleState.getStepInfo()));
```

Comment:
I need to go back and do something else here because this is really ugly, but otherwise it throws NPEs and I wanted to get eyes on the rest of the code.